src/main/scala/is/hail/expr/AST.scala

```diff
@@ -289,6 +290,8 @@ sealed abstract class AST(pos: Position, subexprs: Array[AST] = Array.empty) {
       f(values)
     }
   }
+
+  def toIR: Option[IR]
```

Comment:
When do we choose `def toIR: Option[IR]` vs `def toIR: PartialFunction[(), IR]`?

Reply:
In simple cases, the latter leads to nicer syntax, but that same syntax doesn't compose well with `for`-syntax (which arises in more complex circumstances). I assume the latter is less likely to compile into efficient bytecode since it's not a real method but a closure stored as a field (assuming `val toIR: PartialFunction[(), IR]`). Then again, the JVM's JIT has consistently proven my assumptions wrong.